pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -36,9 +36,35 @@ const (
 )
 
 var (
-	defaultReleaseImage = "registry.svc.ci.openshift.org/openshift/origin-release:v4.0"
+	// defaultReleaseImageOriginal is the value served when defaultReleaseImagePadded is unmodified.
+	defaultReleaseImageOriginal = "registry.svc.ci.openshift.org/openshift/origin-release:v4.0"
+	defaultReleaseImagePrefix   = ""
+	// defaultReleaseImagePadded may be replaced in the binary with a pull spec that overrides defaultReleaseImage as
+	// a null-terminated string within the allowed character length. This allows a distributor to override the payload
+	// location without having to rebuild the source.
+	defaultReleaseImagePadded = "_RELEASE_IMAGE_LOCATION_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+	defaultReleaseImageLength = len(defaultReleaseImagePadded)
 )
 
+// defaultReleaseImage abstracts how the binary loads the default release payload. We want to lock the binary
+// to the
+func defaultReleaseImage() string {
+	if strings.HasPrefix(defaultReleaseImagePadded, defaultReleaseImagePrefix) {
+		logrus.Debug("Using default location for release image")
+		return defaultReleaseImageOriginal
+	}
+	nullTerminator := strings.IndexByte(defaultReleaseImagePadded, '\x00')
+	if nullTerminator == -1 {
+		panic(fmt.Sprintf("Release image location was replaced but without a null terminator before %d bytes", defaultReleaseImageLength))
+	}
+	if nullTerminator > len(defaultReleaseImagePadded) {
+		panic("Release image location contains no null-terminator and constant is corrupted")
```

Comment:
I don't think that this is possible, with Go storing static-string lengths in some separate table.  But I'm fine with an impossible `panic` just in case ;).

Reply:
yeah on the off chance we go to go 1.1x and all hell breaks loose.